bug: image building on M3 macOS failing

Steps taken:

1. Create a venv, `python3 -m venv .warnet`
2. Activate and install warnet, `pip install warnet`
3. Install docker, `brew install --cask docker`
4. Run the image build command, `warcli image build --repo=bitcoin/bitcoin --commit-sha=99ecb9a630e68f7fa5a449065020292790e3e7fb --registry=josibake/warnet --tag master --action=push`

Image starts to build and eventually fails with the following output:

```
148.8   CXX      wallet/libbitcoin_wallet_a-walletutil.o
160.3 g++: fatal error: Killed signal terminated program cc1plus
160.3 compilation terminated.
160.4 make[2]: *** [Makefile:10298: libbitcoin_node_a-init.o] Error 1
160.4 make[2]: *** Waiting for unfinished jobs....
183.6 g++: fatal error: Killed signal terminated program cc1plus
183.6 compilation terminated.
183.6 make[2]: *** [Makefile:10438: libbitcoin_node_a-net_processing.o] Error 1
228.4 make[2]: Leaving directory '/build/bitcoin/src'
228.4 make[1]: *** [Makefile:20311: all-recursive] Error 1
228.4 make[1]: Leaving directory '/build/bitcoin/src'
228.4 make: *** [Makefile:749: all-recursive] Error 1
------
Dockerfile:34
--------------------
  33 |
  34 | >>> RUN set -ex \
  35 | >>>     && cd /build \
  36 | >>>     && git clone --depth 1 "https://github.com/${REPO}" \
  37 | >>>     && cd bitcoin \
  38 | >>>     && git fetch --depth 1 origin "$COMMIT_SHA" \
  39 | >>>     && git checkout "$COMMIT_SHA" \
  40 | >>>     && git apply /tmp/isroutable.patch \
  41 | >>>     && git apply /tmp/addrman.patch \
  42 | >>>     && sed -i s:sys/fcntl.h:fcntl.h: src/compat/compat.h \
  43 | >>>     && ./autogen.sh \
  44 | >>>     && ./configure \
  45 | >>>     LDFLAGS=-L`ls -d /opt/db*`/lib/ \
  46 | >>>     CPPFLAGS=-I`ls -d /opt/db*`/include/ \
  47 | >>>     --prefix=${BITCOIN_PREFIX} \
  48 | >>>     ${BUILD_ARGS} \
  49 | >>>     && make -j$(nproc) \
  50 | >>>     && make install \
  51 | >>>     && strip ${BITCOIN_PREFIX}/bin/bitcoin-cli \
  52 | >>>     && strip ${BITCOIN_PREFIX}/bin/bitcoind \
  53 | >>>     && rm -f ${BITCOIN_PREFIX}/lib/libbitcoinconsensus.a \
  54 | >>>     && rm -f ${BITCOIN_PREFIX}/lib/libbitcoinconsensus.so.0.0.0 \
  55 | >>>     && rm ${BITCOIN_PREFIX}/bin/bitcoin-tx \
  56 | >>>     && rm ${BITCOIN_PREFIX}/bin/bitcoin-wallet \
  57 | >>>     && rm ${BITCOIN_PREFIX}/bin/bitcoin-util
  58 |
--------------------
ERROR: failed to solve: process "/bin/sh -c set -ex     && cd /build     && git clone --depth 1 \"https://github.com/${REPO}\"     && cd bitcoin     && git fetch --depth 1 origin \"$COMMIT_SHA\"     && git checkout \"$COMMIT_SHA\"     && git apply /tmp/isroutable.patch     && git apply /tmp/addrman.patch     && sed -i s:sys/fcntl.h:fcntl.h: src/compat/compat.h     && ./autogen.sh     && ./configure     LDFLAGS=-L`ls -d /opt/db*`/lib/     CPPFLAGS=-I`ls -d /opt/db*`/include/     --prefix=${BITCOIN_PREFIX}     ${BUILD_ARGS}     && make -j$(nproc)     && make install     && strip ${BITCOIN_PREFIX}/bin/bitcoin-cli     && strip ${BITCOIN_PREFIX}/bin/bitcoind     && rm -f ${BITCOIN_PREFIX}/lib/libbitcoinconsensus.a     && rm -f ${BITCOIN_PREFIX}/lib/libbitcoinconsensus.so.0.0.0     && rm ${BITCOIN_PREFIX}/bin/bitcoin-tx     && rm ${BITCOIN_PREFIX}/bin/bitcoin-wallet     && rm ${BITCOIN_PREFIX}/bin/bitcoin-util" did not complete successfully: exit code: 2
bitcoind-builder removed
Buildx builder removed successfully.
```

Interestingly, I sometimes hit a different error (altho this seems intermittent):

```
49.41 config.status: executing depfiles commands
53.72 config.status: error: in '/build/bitcoin':
53.72 config.status: error: Something went wrong bootstrapping makefile fragments
53.72     for automatic dependency tracking.  If GNU make was not used, consider
53.72     re-running the configure script with MAKE="gmake" (or whatever is
53.72     necessary).  You can also try re-running configure with the
53.72     '--disable-dependency-tracking' option to at least be able to build
53.72     the package (albeit without support for automatic dependency tracking).
53.72 See 'config.log' for more details
```

And this won't be fixed until a restart of docker desktop, at which point the original error comes back.

--- a/resources/images/bitcoin/Dockerfile
+++ b/resources/images/bitcoin/Dockerfile
@@ -43,7 +43,7 @@ RUN set -ex \
     && ./autogen.sh \
     && ./configure \
     LDFLAGS=-L`ls -d /opt/db*`/lib/ \
-    CPPFLAGS=-I`ls -d /opt/db*`/include/ \
+    CPPFLAGS="-g0 -I`ls -d /opt/db*`/include/ --param ggc-min-expand=1 --param ggc-min-heapsize=32768" \
     --prefix=${BITCOIN_PREFIX} \
     ${BUILD_ARGS} \
     && make -j$(nproc) \

